WebKit/WebKit
# [ANGLE] Add target for metal_translator_fuzzer

#### 704290fdd90ed38cc649e7b8623115fb4260c9a6
<pre>
[ANGLE] Add target for metal_translator_fuzzer
<a href="https://bugs.webkit.org/show_bug.cgi?id=268201">https://bugs.webkit.org/show_bug.cgi?id=268201</a>
&lt;<a href="https://rdar.apple.com/98947215">rdar://98947215</a>&gt;

Reviewed by Kimmo Kinnunen.

* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
- Add BaseTarget.xcconfig and metal_translator_fuzzer.xcconfig.
(metal_translator_fuzzer): Add target.
- Add translator_fuzzer.cpp to the project.
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
- Add copyright header.
(ANGLE_ALLOWABLE_CLIENTS): Add.
- Disable -allowable_client switches when libFuzzer is enabled.
(GCC_SYMBOLS_PRIVATE_EXTERN): Add.
- Disable private extern symbols when libFuzzer is enabled.
* Source/ThirdParty/ANGLE/Configurations/BaseTarget.xcconfig: Add.
- Extract common settings from ANGLE-dynamic.xcconfig.
(ANGLE_HEADER_PATH_PREFIX): Add.
- Add variable for header include path prefix so it may be overridden.
* Source/ThirdParty/ANGLE/Configurations/metal_translator_fuzzer.xcconfig: Add.
* Source/ThirdParty/ANGLE/src/compiler/fuzz/translator_fuzzer.cpp:
(LLVMFuzzerTestOneInput):
- Add a way to disable non-Metal outputs at build time.
- Add SH_MSL_METAL_OUTPUT as a valid output.
- Call sh::Finalize() after each test case since sh::Initialize is
  called at the start of every test case.

Canonical link: <a href="https://commits.webkit.org/273682@main">https://commits.webkit.org/273682@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/b975b63454221a79706fbf0d73febc732e17ad7c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36357 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15311 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38576 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/39074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/32672 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17755 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12342 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/39074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36917 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/12924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/32233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/39074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/32450 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/40318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/32989 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/32812 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/40318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/11565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/9445 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/40318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/12017 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4710 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/12425 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | |
